### DISCUSSION TITLE
AMQP 1.0 shovel can't able to handle messages

### DISCUSSION THREAD

**Discussion by saitejarsv** [rabbitmq/rabbitmq-server#8794] [Other]
### Is your feature request related to a problem? Please describe.

Hello,

We are running Rabbitmq 3.12.0 in a windows machine and enabled the AMQP1.0. 
Our motto is to integrate this rabbitq to IBM mq 9.1.0 which is having AMQP 1.0 connection channel.

We created the shovels and those are successfully able to establish the connection with IBM MQ.
But when we try to send a message from IBM to Rabbitmq, Rabbitmq got the error logs stating, Virtual host cant able to handle the destination message. 

2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>          {rabbit_framing_amqp_0_9_1,shortstr_size,1,
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>                                     [{file,"rabbit_framing_amqp_0_9_1.erl"},
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>                                      {line,178}]},
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>          {rabbit_framing_amqp_0_9_1,encode_method_fields,1,
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>                                     [{file,"rabbit_framing_amqp_0_9_1.erl"},
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>                                      {line,977}]},
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>          {rabbit_binary_generator,build_simple_method_frame,3,
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>                                   [{file,"rabbit_binary_generator.erl"},
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>                                    {line,50}]},
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>          {rabbit_writer,assemble_frames,5,
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>                         [{file,"rabbit_writer.erl"},{line,329}]},
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>          {rabbit_writer,internal_send_command_async,3,
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>                         [{file,"rabbit_writer.erl"},{line,362}]},
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>          {rabbit_writer,handle_message,3,
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>                         [{file,"rabbit_writer.erl"},{line,229}]},
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>          {rabbit_writer,mainloop1,2,[{file,"rabbit_writer.erl"},{line,213}]}]}
2023-07-06 11:59:25.915000+02:00 [error] <0.740.0> 
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>     supervisor: {<0.737.0>,amqp_channel_sup}
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>     errorContext: child_terminated
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>     reason: {badarg,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                 [{erlang,size,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                      [undefined],
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                      [{error_info,#{module => erl_erts_errors}}]},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                  {rabbit_framing_amqp_0_9_1,shortstr_size,1,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                      [{file,"rabbit_framing_amqp_0_9_1.erl"},{line,178}]},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                  {rabbit_framing_amqp_0_9_1,encode_method_fields,1,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                      [{file,"rabbit_framing_amqp_0_9_1.erl"},{line,977}]},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                  {rabbit_binary_generator,build_simple_method_frame,3,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                      [{file,"rabbit_binary_generator.erl"},{line,50}]},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                  {rabbit_writer,assemble_frames,5,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                      [{file,"rabbit_writer.erl"},{line,329}]},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                  {rabbit_writer,internal_send_command_async,3,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                      [{file,"rabbit_writer.erl"},{line,362}]},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                  {rabbit_writer,handle_message,3,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                      [{file,"rabbit_writer.erl"},{line,229}]},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                  {rabbit_writer,mainloop1,2,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                      [{file,"rabbit_writer.erl"},{line,213}]}]}
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>     offender: [{pid,<0.740.0>},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                {id,writer},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                {mfargs,{rabbit_writer,start_link,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                                       [#Port<0.36>,1,131072,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                                        rabbit_framing_amqp_0_9_1,<0.739.0>,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                                        {<<"client 127.0.0.1:48975 -> 127.0.0.1:5673">>,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                                         1},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                                        false,1000000000]}},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                {restart_type,transient},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                {significant,false},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                {shutdown,300000},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                {child_type,worker}]
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> 
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>     supervisor: {<0.737.0>,amqp_channel_sup}
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>     errorContext: shutdown
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>     reason: reached_max_restart_intensity
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>     offender: [{pid,<0.740.0>},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                {id,writer},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                {mfargs,{rabbit_writer,start_link,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                                       [#Port<0.36>,1,131072,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                                        rabbit_framing_amqp_0_9_1,<0.739.0>,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                                        {<<"client 127.0.0.1:48975 -> 127.0.0.1:5673">>,
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                                         1},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                                        false,1000000000]}},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                {restart_type,transient},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                {significant,false},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                {shutdown,300000},
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>                {child_type,worker}]
2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> 
2023-07-06 11:59:25.916000+02:00 [warning] <0.631.0> Shovel 'IBM2rabbit' in virtual host '/' could not handle a destination message {'EXIT',
2023-07-06 11:59:25.916000+02:00 [warning] <0.631.0>                                                                                 <0.739.0>,
2023-07-06 11:59:25.916000+02:00 [warning] <0.631.0>                                                                                 shutdown}

### Describe the solution you'd like

May be Something needs to done at AMQP 1.0 plugin or in the Elang modules

### Describe alternatives you've considered

I used both True and False probabilities in the config files for this parameters.

 

amqp1_0.map_subject_to_routing_key = true | false

 

amqp1_0.copy_subject_to_amqp091_headers = true | false

 

Both gives the same result

### Additional context

We are trying to exchange the messages between 2 MQs.

The first is a legacy system in a different organization which is only able to exchange with IBM MQ. Currently that IBM mq is talking to target mq in our environment IBM MQ.

And now, we want to change the target MQ to Rabbit MQ for the integration.

**Comment by kjnilsson** [comment]
It looks like somehow a string that is too long is ending up as a shortstr.
It isn't clear at all how that happened but perhaps if you could share all
the AMQP 1.0 headers and message annotations from the IBM MQ message then
we might be able to work something out.

On Thu, 6 Jul 2023 at 11:35, saitejarsv ***@***.***> wrote:

> Is your feature request related to a problem? Please describe.
>
> Hello,
>
> We are running Rabbitmq 3.12.0 in a windows machine and enabled the
> AMQP1.0.
> Our motto is to integrate this rabbitq to IBM mq 9.1.0 which is having
> AMQP 1.0 connection channel.
>
> We created the shovels and those are successfully able to establish the
> connection with IBM MQ.
> But when we try to send a message from IBM to Rabbitmq, Rabbitmq got the
> error logs stating, Virtual host cant able to handle the destination
> message.
>
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> {rabbit_framing_amqp_0_9_1,shortstr_size,1,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> [{file,"rabbit_framing_amqp_0_9_1.erl"},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0> {line,178}]},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> {rabbit_framing_amqp_0_9_1,encode_method_fields,1,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> [{file,"rabbit_framing_amqp_0_9_1.erl"},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0> {line,977}]},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> {rabbit_binary_generator,build_simple_method_frame,3,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> [{file,"rabbit_binary_generator.erl"},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0> {line,50}]},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> {rabbit_writer,assemble_frames,5,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> [{file,"rabbit_writer.erl"},{line,329}]},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> {rabbit_writer,internal_send_command_async,3,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> [{file,"rabbit_writer.erl"},{line,362}]},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> {rabbit_writer,handle_message,3,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> [{file,"rabbit_writer.erl"},{line,229}]},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> {rabbit_writer,mainloop1,2,[{file,"rabbit_writer.erl"},{line,213}]}]}
> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> supervisor:
> {<0.737.0>,amqp_channel_sup}
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> errorContext:
> child_terminated
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> reason: {badarg,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> [{erlang,size,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> [undefined],
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> [{error_info,#{module
> => erl_erts_errors}}]},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> {rabbit_framing_amqp_0_9_1,shortstr_size,1,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> [{file,"rabbit_framing_amqp_0_9_1.erl"},{line,178}]},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> {rabbit_framing_amqp_0_9_1,encode_method_fields,1,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> [{file,"rabbit_framing_amqp_0_9_1.erl"},{line,977}]},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> {rabbit_binary_generator,build_simple_method_frame,3,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> [{file,"rabbit_binary_generator.erl"},{line,50}]},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> {rabbit_writer,assemble_frames,5,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> [{file,"rabbit_writer.erl"},{line,329}]},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> {rabbit_writer,internal_send_command_async,3,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> [{file,"rabbit_writer.erl"},{line,362}]},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> {rabbit_writer,handle_message,3,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> [{file,"rabbit_writer.erl"},{line,229}]},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> {rabbit_writer,mainloop1,2,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> [{file,"rabbit_writer.erl"},{line,213}]}]}
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> offender:
> [{pid,<0.740.0>},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {id,writer},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> {mfargs,{rabbit_writer,start_link,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> [#Port<0.36>,1,131072,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> rabbit_framing_amqp_0_9_1,<0.739.0>,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {<<"client
> 127.0.0.1:48975 -> 127.0.0.1:5673">>,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> 1},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> false,1000000000]}},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> {restart_type,transient},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {significant,false},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {shutdown,300000},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {child_type,worker}]
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> supervisor:
> {<0.737.0>,amqp_channel_sup}
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> errorContext: shutdown
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> reason:
> reached_max_restart_intensity
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> offender:
> [{pid,<0.740.0>},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {id,writer},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> {mfargs,{rabbit_writer,start_link,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> [#Port<0.36>,1,131072,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> rabbit_framing_amqp_0_9_1,<0.739.0>,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {<<"client
> 127.0.0.1:48975 -> 127.0.0.1:5673">>,
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> 1},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> false,1000000000]}},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> {restart_type,transient},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {significant,false},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {shutdown,300000},
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {child_type,worker}]
> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
> 2023-07-06 11:59:25.916000+02:00 [warning] <0.631.0> Shovel 'IBM2rabbit'
> in virtual host '/' could not handle a destination message {'EXIT',
> 2023-07-06 11:59:25.916000+02:00 [warning] <0.631.0> <0.739.0>,
> 2023-07-06 11:59:25.916000+02:00 [warning] <0.631.0> shutdown}
> Describe the solution you'd like
>
> May be Something needs to done at AMQP 1.0 plugin or in the Elang modules
> Describe alternatives you've considered
>
> I used both True and False probabilities in the config files for this
> parameters.
>
> amqp1_0.map_subject_to_routing_key = true | false
>
> amqp1_0.copy_subject_to_amqp091_headers = true | false
>
> Both gives the same result
> Additional context
>
> We are trying to exchange the messages between 2 MQs.
>
> The first is a legacy system in a different organization which is only
> able to exchange with IBM MQ. Currently that IBM mq is talking to target mq
> in our environment IBM MQ.
>
> And now, we want to change the target MQ to Rabbit MQ for the integration.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rabbitmq/rabbitmq-server/issues/8793>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJAHFDPHWL2ZSGJFB5EFK3XO2IIBANCNFSM6AAAAAA2AHPBMY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
*Karl Nilsson*

**Comment by kjnilsson** [comment]
In addition server logs and the full shovel configuration would be helpful.

On Thu, 6 Jul 2023 at 12:25, Karl Nilsson ***@***.***> wrote:

> It looks like somehow a string that is too long is ending up as a
> shortstr. It isn't clear at all how that happened but perhaps if you could
> share all the AMQP 1.0 headers and message annotations from the IBM MQ
> message then we might be able to work something out.
>
> On Thu, 6 Jul 2023 at 11:35, saitejarsv ***@***.***> wrote:
>
>> Is your feature request related to a problem? Please describe.
>>
>> Hello,
>>
>> We are running Rabbitmq 3.12.0 in a windows machine and enabled the
>> AMQP1.0.
>> Our motto is to integrate this rabbitq to IBM mq 9.1.0 which is having
>> AMQP 1.0 connection channel.
>>
>> We created the shovels and those are successfully able to establish the
>> connection with IBM MQ.
>> But when we try to send a message from IBM to Rabbitmq, Rabbitmq got the
>> error logs stating, Virtual host cant able to handle the destination
>> message.
>>
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> {rabbit_framing_amqp_0_9_1,shortstr_size,1,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> [{file,"rabbit_framing_amqp_0_9_1.erl"},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0> {line,178}]},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> {rabbit_framing_amqp_0_9_1,encode_method_fields,1,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> [{file,"rabbit_framing_amqp_0_9_1.erl"},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0> {line,977}]},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> {rabbit_binary_generator,build_simple_method_frame,3,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> [{file,"rabbit_binary_generator.erl"},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0> {line,50}]},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> {rabbit_writer,assemble_frames,5,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> [{file,"rabbit_writer.erl"},{line,329}]},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> {rabbit_writer,internal_send_command_async,3,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> [{file,"rabbit_writer.erl"},{line,362}]},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> {rabbit_writer,handle_message,3,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> [{file,"rabbit_writer.erl"},{line,229}]},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> {rabbit_writer,mainloop1,2,[{file,"rabbit_writer.erl"},{line,213}]}]}
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.740.0>
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> supervisor:
>> {<0.737.0>,amqp_channel_sup}
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> errorContext:
>> child_terminated
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> reason: {badarg,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> [{erlang,size,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> [undefined],
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> [{error_info,#{module
>> => erl_erts_errors}}]},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> {rabbit_framing_amqp_0_9_1,shortstr_size,1,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> [{file,"rabbit_framing_amqp_0_9_1.erl"},{line,178}]},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> {rabbit_framing_amqp_0_9_1,encode_method_fields,1,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> [{file,"rabbit_framing_amqp_0_9_1.erl"},{line,977}]},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> {rabbit_binary_generator,build_simple_method_frame,3,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> [{file,"rabbit_binary_generator.erl"},{line,50}]},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> {rabbit_writer,assemble_frames,5,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> [{file,"rabbit_writer.erl"},{line,329}]},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> {rabbit_writer,internal_send_command_async,3,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> [{file,"rabbit_writer.erl"},{line,362}]},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> {rabbit_writer,handle_message,3,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> [{file,"rabbit_writer.erl"},{line,229}]},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> {rabbit_writer,mainloop1,2,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> [{file,"rabbit_writer.erl"},{line,213}]}]}
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> offender:
>> [{pid,<0.740.0>},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {id,writer},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> {mfargs,{rabbit_writer,start_link,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> [#Port<0.36>,1,131072,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> rabbit_framing_amqp_0_9_1,<0.739.0>,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {<<"client
>> 127.0.0.1:48975 -> 127.0.0.1:5673">>,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> 1},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> false,1000000000]}},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> {restart_type,transient},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {significant,false},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {shutdown,300000},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {child_type,worker}]
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> supervisor:
>> {<0.737.0>,amqp_channel_sup}
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> errorContext: shutdown
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> reason:
>> reached_max_restart_intensity
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> offender:
>> [{pid,<0.740.0>},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {id,writer},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> {mfargs,{rabbit_writer,start_link,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> [#Port<0.36>,1,131072,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> rabbit_framing_amqp_0_9_1,<0.739.0>,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {<<"client
>> 127.0.0.1:48975 -> 127.0.0.1:5673">>,
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> 1},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> false,1000000000]}},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> {restart_type,transient},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {significant,false},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {shutdown,300000},
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0> {child_type,worker}]
>> 2023-07-06 11:59:25.915000+02:00 [error] <0.737.0>
>> 2023-07-06 11:59:25.916000+02:00 [warning] <0.631.0> Shovel 'IBM2rabbit'
>> in virtual host '/' could not handle a destination message {'EXIT',
>> 2023-07-06 11:59:25.916000+02:00 [warning] <0.631.0> <0.739.0>,
>> 2023-07-06 11:59:25.916000+02:00 [warning] <0.631.0> shutdown}
>> Describe the solution you'd like
>>
>> May be Something needs to done at AMQP 1.0 plugin or in the Elang modules
>> Describe alternatives you've considered
>>
>> I used both True and False probabilities in the config files for this
>> parameters.
>>
>> amqp1_0.map_subject_to_routing_key = true | false
>>
>> amqp1_0.copy_subject_to_amqp091_headers = true | false
>>
>> Both gives the same result
>> Additional context
>>
>> We are trying to exchange the messages between 2 MQs.
>>
>> The first is a legacy system in a different organization which is only
>> able to exchange with IBM MQ. Currently that IBM mq is talking to target mq
>> in our environment IBM MQ.
>>
>> And now, we want to change the target MQ to Rabbit MQ for the integration.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/rabbitmq/rabbitmq-server/issues/8793>, or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAJAHFDPHWL2ZSGJFB5EFK3XO2IIBANCNFSM6AAAAAA2AHPBMY>
>> .
>> You are receiving this because you are subscribed to this thread.Message
>> ID: ***@***.***>
>>
>
>
> --
> *Karl Nilsson*
>


-- 
*Karl Nilsson*

**Reply by saitejarsv** [reply]
I just sent a text message in manual manner to test. So no headers and annotations.

Below is the Shovel configuration out put for "rabbitmqctl shovel_status" (masked the credentials)

Shovel status on node rabbit@SRVWUIBAMLEST01

destination         destination_address       destination_protocol              last_changed     name    source  source_exchange              source_exchange_key    source_protocol              state              termination_reason       type      vhost

amqp://####:####@localhost:5672        localhost             amqp10              2023-07-06 11:58:16     

rabbit2ibm         amqp://localhost:5673  outgoing                           amqp091              running               dynamic             /

amqp://localhost:5673  incoming            amqp091            2023-07-06 12:50:12            

IBM2rabbit        amqp://###:###@localhost:5672            localhost              amqp10              running               dynamic             /

  
FYI::

I'm running the rabitmq on 5673 port, as 5672 is already being used by AMQP of IBM

**Reply by michaelklishin** [reply]
We do not have access to commercial IBM products and "just run IBM MQ" is not a reproduction step
for over 99% of the RabbitMQ community.

Please take a [traffic capture](https://rabbitmq.com/amqp-wireshark.html) using tcpdump (or Wireshark if this can be reproduced locally without containers), then [share it privately with out team](https://rabbitmq.com/contact.html).

If we don't know what IBM MQ sends *exactly*, we won't be able to do something about this behavior.

**Reply by saitejarsv** [reply]
Shared the Wireshark dup in private to the "rabbitmq-core@groups.vmware.com". Please check that and let me know if you required anything further.

**Comment by michaelklishin** [comment]
The exception comes from the AMQP 0-9-1 deserialization module. So the source is an AMQP 1.0 message and the destination is an AMQP 0-9-1 target (for example, an exchange)

We need an executable way to reproduce, or at the very least, a [traffic capture](https://rabbitmq.com/amqp-wireshark.html) from the RabbitMQ node.

**Comment by saitejarsv** [comment]
I just sent a text message in manual manner to test. So no headers and annotations.

Below is the Shovel configuration out put for "rabbitmqctl shovel_status" (masked the credentials)

Shovel status on node rabbit@SRVWUIBAMLEST01

destination         destination_address       destination_protocol              last_changed     name    source  source_exchange              source_exchange_key    source_protocol              state              termination_reason       type      vhost

amqp://####:####@localhost:5672        localhost             amqp10              2023-07-06 11:58:16     

rabbit2ibm         amqp://localhost:5673  outgoing                           amqp091              running               dynamic             /

amqp://localhost:5673  incoming            amqp091            2023-07-06 12:50:12            

IBM2rabbit        amqp://###:###@localhost:5672            localhost              amqp10              running               dynamic             /

  
FYI::

I'm running the rabitmq on 5673 port, as 5672 is already being used by AMQP of IBM

**Reply by michaelklishin** [reply]
We do not have access to commercial IBM products and "just run IBM MQ" is not a reproduction step
for over 99% of the RabbitMQ community.

Please take a [traffic capture](https://rabbitmq.com/amqp-wireshark.html) using tcpdump (or Wireshark if this can be reproduced locally without containers), then [share it privately with out team](https://rabbitmq.com/contact.html).

If we don't know what IBM MQ sends *exactly*, we won't be able to do something about this behavior.

**Comment by michaelklishin** [comment]
We have received a traffic capture from @saitejarsv. Some relevant bits might be posted here, we are assessing if there's any sensitive data.

Since this is a high PTO season, it can take a while.

**Reply by saitejarsv** [reply]
Hello Michael, 

Any update on this request..? We are waiting on the resolution for 2 months.

**Reply by michaelklishin** [reply]
There are no updates. RabbitMQ is open source software and our team is small. You are welcome to investigate this behavior and submit a PR if this is something urgent for you. I don't have an ETA as to when we may get to investigate this, possibly as part of #9022 for 4.0.
